srv/salt/ceph/rebuild/default.sls

```diff
@@ -0,0 +1,93 @@
+
+{% set master = salt['pillar.get']('master_minion') %}
+
+{% if salt['saltutil.runner']('disengage.check', cluster='ceph') == False %}
+safety is engaged:
+  salt.state:
+    - tgt: {{ master }}
+    - name: "Run 'salt-run disengage.safety' to disable"
+    - failhard: True
+
+{% endif %}
+
+wait on healthy cluster:
+  salt.state:
+    - tgt: {{ master }}
+    - tgt_type: compound
+    - sls: ceph.wait.until.OK
+    - failhard: True
+
+{% if salt['saltutil.runner']('filequeue.empty', queue='rebuild') %}
+{% for host in salt.saltutil.runner('select.minions', cluster='ceph', roles='storage') %}
+queue {{ host }}:
+  salt.runner:
+    - name: filequeue.add
+    - queue: 'rebuild'
+    - item: {{ host }}
+
+{% endfor %}
+
+advise rebuild:
+  salt.runner:
+    - name: advise.rebuild
+
+{% endif %}
+
+{% for host in salt['saltutil.runner']('filequeue.array', queue='rebuild') %}
+lock obliterate {{ host }}:
+  salt.state:
+    - tgt: {{ host }}
+    - tgt_type: compound
+    - sls: ceph.obliterate.lock
```

Comment:
Yeah, it would be redundant.  I just verified with a small example.  I think showing the "remove osds" will not execute in red because the lock failed helps get the point across.